tests/callbacks/test_model_checkpoint.py

```diff
@@ -250,8 +251,28 @@ def test_model_checkpoint_none_monitor(tmpdir):
     assert checkpoint_callback.kth_best_model_path == ''
 
     # check that the correct ckpts were created
-    expected = ['lightning_logs']
-    expected.extend(f'epoch={e}.ckpt' for e in range(epochs))
+    expected = [f'epoch={e}.ckpt' for e in range(epochs)]
+    assert set(os.listdir(tmpdir)) == set(expected)
+
+
+@pytest.mark.parametrize("period", list(range(4)))
+def test_model_checkpoint_period(tmpdir, period):
+    model = EvalModelTemplate()
+    epochs = 5
+    checkpoint_callback = ModelCheckpoint(filepath=tmpdir, save_top_k=-1, period=period)
```

Comment:
```suggestion
    checkpoint_callback = ModelCheckpoint(filepath=tmpdir, save_top_k=-1, monitor="val_loss", period=period)
```
need this, or the test fails on master for the wrong reason

Reply:
monitor with save_top_k is now required, there is no default monitor anymore.

Reply:
There was an error due to an extra comma. Ive fixed it and tests are running again. No errors yet.

We actually want to always save in that test, `monitor=None`

Reply:
what about the error on master? Instead of your assertions failing, I see a misconfiguration error telling me the monitor is required.

Reply:
[this change](https://github.com/PyTorchLightning/pytorch-lightning/pull/3630/files#diff-63eb50f09b915db4167aed7cb38ac5c0R218) is avoiding the check. As per the monitor docstring, it should check if top_k != 0 or 1

Reply:
ok, I think I understand partially, but could you check my comment below too?

Reply:
Question is:

With https://github.com/PyTorchLightning/pytorch-lightning/commit/d79bce1dffcc2d65799987608afb4a97d3090451 changes

How do you always save a checkpoint?

Reply:
ah it's annoying that save_top_k=1 is the default. it would suggest that we are saving the best but we are not right now. 
with monitor=None as default, I think save_top_k should also be None as default.

Reply:
IMO top_k should be -1 by default.

Reply:
no I disagree, that would save too many files

Reply:
what's the point of having `monitor=None` if it is meant to save a checkpoint only the last epoch??

Reply:
Current default behaviour does the same as save_last, right? But with a different checkpoint name.

In any case, I think we can land this PR and discuss that in a different PR